pandas/core/sparse/array.py

```diff
@@ -292,9 +292,9 @@ def __setstate__(self, state):
         self._fill_value = fill_value
 
     def __len__(self):
-        try:
+        if hasattr(self, 'sp_index'):
```

Comment:
This felt really icky to me to try and catch something... Feel free to push back on that change.

Reply:
How was this failing before?

Reply:
I have no clue! :smile: I couldn't figure out why the try / except was there in the first place tbh.

Reply:
huh?

Reply:
This isn't essential to this PR. I just don't understand why this code was here in the first place.

Reply:
Meaning I can take it out to bring down the code changes